src/server/templates/zod.ts

```diff
@@ -0,0 +1,323 @@
+    // Array
+    if (pgType.startsWith("_")) {
+        return basicZodType(pgType.substring(1)) + ".array()"
+    }
+
+    if ([ 'bool', 'boolean' ].includes(pgType)) {
+        return "boolean()"
+    }
+
+    if (['int2', 'int4', 'int8', 'float4', 'float8', 'numeric', 'integer', 'bigint', 'oid'].includes(pgType)) {
+        return 'number()'
+    }
+
+    if (
+        [
+            'bytea',
+            'bpchar',
+            'varchar',
+            'text',
+            'citext',
+            'uuid',
+            'vector',
+            'json',
+            'jsonb',
+            'inet',
+            'cidr',
+            'macaddr',
+            'macaddr8',
+            'character varying',
+        ].includes(pgType)
+    ) {
+        return 'string()'
+    }
+
+    if (["date", "time", "timetz", "timestamp", "timestamptz", "timestamp with time zone"].includes(pgType)) {
+        return 'date()'
+    }
+
+    console.debug(`Basic Zod Type: Unknown type ${pgType}`)
+
+    // Everything else is an enum
+    return "string()"
+}
+
+const IP_REGEX = "^((((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))|((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))))\\/[0-9]{1,3}$"
+const UUID_REGEX = "^[0-9a-fA-F]{8}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{4}\\b-[0-9a-fA-F]{12}$"
+
+function extractExtraZodMethods(column: PostgresColumn): string[] {
+    const methods: string[] = []
+
+    // UUID
+    if (column.format === "uuid") {
+        methods.push(`regex(/${UUID_REGEX}/)`)
+
+        if (column.default_value === "gen_random_uuid()") {
+            methods.push("default(uuidv4)")
+        }
+    }
+
+    // Date and time types
+    if (["date", "time", "timetz", "timestamp", "timestamptz"].includes(column.format)) {
+        if (column.default_value === "now()") {
+            methods.push("default(() => new Date())")
+        }
+    }
+
+    // Enums
+    if (column.data_type === "USER-DEFINED") {
+        methods.push(`enum([${column.enums.map((value) => `"${value}"`).join(', ')}] as const)`)
+    }
+
+    if (column.format === "inet") {
+        // Zods `ip` method doesn't check for subnets, so we use our own regex instead.
+        methods.push(`regex(/${IP_REGEX}/)`)
+    }
+
+    return methods
+}
+
+function extractGeneralZodMethods(column: PostgresColumn): string[] {
```

Comment:
lets rename it to `extractNullabesAndOptionals` to something better than "general"

Reply:
that's too specific imo

Reply:
that's too specific in my opinion